### ISSUE TITLE
Incompatible with React Native >=0.7.1: More than one plugin attempted to publish the same registration name, `onError`

### ISSUE THREAD

**Issue by jevakallio** [paramaggarwal/react-native-youtube#6]
The application won't start due to the error, as in title:
![screen shot 2015-08-03 at 17 43 58](https://cloud.githubusercontent.com/assets/1203949/9043041/9bbf77a2-3a0a-11e5-92c5-589711d53dd0.png)

This is because react native `RCTImageViewManager` already publishes an event `onError`, for discussion see: https://github.com/facebook/react-native/issues/1952

I have a solution to this in my fork, will test it works and submit a PR to fix.

**Comment by jevakallio** [comment]
Submitted PR to fix this issue. 

Based on empirical testing, merging https://github.com/paramaggarwal/react-native-youtube/pull/5 and https://github.com/paramaggarwal/react-native-youtube/pull/7 make this component compatible at least up to React Native 0.9.

**Comment by paramaggarwal** [comment]
Thanks a lot @jevakallio! Merged both PRs.

**Comment by KabirKang** [comment]
I am still getting this issue with the onError function for Android.

By the way, does npm install work or does it need to be git cloned?

**Comment by undef1ned** [comment]
@KabirKang I'm not using this module in my app, it's even not listed in package.json / node_modules, however I got this issue as well after updating to rn 0.39. So just googled this issue here :) 

However one of my own modules had this method and disabling the module helped me to build the app

Maybe it's related to this change:
https://github.com/facebook/react-native/commit/e87e181998bf4f8efba2c7e04bec0929344600b9

**Comment by IjzerenHein** [comment]
I'm also seeing this after upgrading to RN 0.39. Not sure it's related to react-native-youtube though. Have not been able to pinpoint the actual problem yet. 

@KabirKang what exactly did you change in your app? Could you post the changes you made in your app to fix this. Maybe this will help the rest of us on how to solve it in our apps. Thanks!!

**Comment by undef1ned** [comment]
@IjzerenHein Simply check your other node_modules/* whether some of them exposes onError method. I did the same changes to my custom module, as it's here https://github.com/inProgress-team/react-native-youtube/issues/7 and the issue disappeared. I think it was related to Image.onError which is defined by RN now.

**Comment by IjzerenHein** [comment]
@undef1ned Thanks for the feedback!

Finally found that it is indeed the `onError` event in `react-native-youtube` that is the culprit (in my app). I don't quite know how to properly fix it, but after renaming the following instance of `"onError"` the error is gone and the app starts. Hope this helps anyone else in actually fixing the problem properly..

```java
// YouTubeManager.java
@Override
	public
	@Nullable
	Map getExportedCustomDirectEventTypeConstants() {
		return MapBuilder.of(
				"error",
				MapBuilder.of("registrationName", "onError2"), // <-- renamed from onError to onError2
				"ready",
				MapBuilder.of("registrationName", "onReady"),
```

**Comment by undef1ned** [comment]
@IjzerenHein Yeah, it's also present in the Youtube.android.js so you should replace it everywhere to make it work and may-be submit pull-request :)

**Comment by IjzerenHein** [comment]
FYI, I submitted a PR for this #90 to fix the issue on RN 0.39
